src/ndarray.jl

```diff
@@ -701,6 +701,50 @@ function /(arg0 :: NDArray, arg :: Real)
   arg0 ./ arg
 end
 
+power(x::NDArray, y::NDArray) = _power(x, y)
+
+power(x::NDArray, s::Real) = _power_scalar(x, scalar=s)
+power(s::Real, x::NDArray) = _rpower_scalar(x, scalar=s)
+
+power(x::NDArray, s::Integer) = _power_scalar(x, scalar=s)
+power(s::Integer, x::NDArray) = _rpower_scalar(x, scalar=s)
+
+import Base: ^
+
+"""
+    ^(x::NDArray, y::NDArray)
+    ^(x::NDArray, s::Real)
+    ^(s::Real, x::NDArray)
+
+Elementwise power
+"""
+^(x::NDArray, y::NDArray) = power(x, y)
+
+^(x::NDArray, s::Real) = power(x, s)
```

Comment:
I don't think this is the semantic we want. In Julia we would get A^2 = A*A which is matrix multiply and not the elementwise power. Now the question is what to do for tensors instead of matrices and if NDArrays are mathematical constructs or containers. In any case I would prefer to only define the broadcasting operation since that is well-defined.

Reply:
I think it's same story: https://github.com/dmlc/MXNet.jl/pull/292/files#r143826115